src/wp-includes/functions.wp-scripts.php

```diff
@@ -166,20 +167,32 @@ function wp_add_inline_script( $handle, $data, $position = 'after' ) {
  *                                    as a query string for cache busting purposes. If version is set to false, a version
  *                                    number is automatically added equal to current installed WordPress version.
  *                                    If set to null, no version is added.
- * @param bool             $in_footer Optional. Whether to enqueue the script before `</body>` instead of in the `<head>`.
- *                                    Default 'false'.
+ * @param array|bool       $args     {
+ *      Optional. An array of additional script loading strategies. Default empty array.
+ *      Otherwise, it may be a boolean in which case it determines whether the script is printed in the footer. Default false.
+ *
+ *      @type string    $strategy     Optional. If provided, may be either 'defer' or 'async'.
+ *      @type bool      $in_footer    Optional. Whether to print the script in the footer. Default 'false'.
+ * }
  * @return bool Whether the script has been registered. True on success, false on failure.
  */
-function wp_register_script( $handle, $src, $deps = array(), $ver = false, $in_footer = false ) {
+function wp_register_script( $handle, $src, $deps = array(), $ver = false, $args = array() ) {
```

Comment:
Named arguments was introduced in PPHP 8.0, so the function can be called in this way `wp_register_script(in_footer: true, handle: 'xyz')`.
In this case this is a breaking change. Should we left `$in_footer` in the arguments instead of changing to `$args`?

Reply:
Or perhaps add a 5th function parameter retaining the `$in_footer` name?

Reply:
Great! That will work in every combination.